set-web/src/main/java/eu/europeana/set/web/service/controller/BaseRest.java

```diff
@@ -202,72 +140,28 @@ protected String serializeUserSet(LdProfiles profile, UserSet storedUserSet) thr
 		return serializedUserSetJsonLdStr;
 	}
 
-	/**
-	 * This method checks timestamp if provided within the "If-Match" HTTP
-	 * header, if false responds with HTTP 412
-	 * 
-	 * @param request
-	 * @param modified
-	 * @throws ApplicationAuthenticationException
-	 */
-	public void checkHeaderTimestamp(HttpServletRequest request, UserSet userSet)
-			throws ApplicationAuthenticationException {
-		int modified = userSet.getModified().hashCode();
+    /**
+     * This method compares If-Match header with the current etag value.
+     * 
+     * @param etag    The current etag value
+     * @param request The request containing If-Match header
+     * @throws HttpException
+     */
+    public void checkIfMatchHeader(int etag, HttpServletRequest request) throws HttpException {
+
 		String ifMatchHeader = request.getHeader(HttpHeaders.IF_MATCH);
 		if (ifMatchHeader != null) {
-			getLogger().trace("'If-Match' header value: " + ifMatchHeader);
-			String modifiedStr = String.valueOf(modified);
-			if (!ifMatchHeader.equals(modifiedStr)) {
-				throw new ApplicationAuthenticationException(I18nConstants.INVALID_IF_MATCH_TIMESTAMP,
-						I18nConstants.INVALID_IF_MATCH_TIMESTAMP, new String[] { modifiedStr },
-						HttpStatus.PRECONDITION_FAILED, null);
-			}
+		    try {
+		    	int ifMatchValue = Integer.parseInt(ifMatchHeader);
+		    	if (etag != ifMatchValue)
+		    		throw new HeaderValidationException(I18nConstants.INVALID_PARAM_VALUE, HttpHeaders.IF_MATCH,
+		    				ifMatchHeader);
+		    } catch (NumberFormatException e) {
+		    	throw new HeaderValidationException(I18nConstants.INVALID_PARAM_VALUE, HttpHeaders.IF_MATCH,
+		    			ifMatchHeader);
+		    }
 		}
-	}
-
-	/**
-	 * This method validates whether user has admin rights to execute methods in
-	 * management API.
-	 * 
-	 * @param apiKey
-	 * @param userToken
-	 * @return true if user has necessary permissions
-	 */
-	protected boolean isAdmin(String apiKey, String userToken) {
-		return (apiKey.equals("apidemo") && userToken.equals("admin"));
-	}
-
-	/**
-	 * This method checks that only the admins and the owners of the user sets
-	 * are allowed to delete the user set. in the case of regular users (not
-	 * admins), the autorization method must check if the users that calls the
-	 * deletion (i.e. identified by provided user token) is the same user as the
-	 * creator of the user set
-	 * 
-	 * @param userSet
-	 * @param wsKey
-	 * @param queryUser
-	 * @throws ApplicationAuthenticationException
-	 */
-	public void hasModifyRights(UserSet userSet, String wsKey, String queryUser)
-			throws OperationAuthorizationException {
-
-		if (!(isAdmin(wsKey, queryUser) || userSet.getCreator().getName().equals(queryUser))) {
```

Comment:
the admin or owner verification is still required here, see the equivalent implementation in the annotation api

Reply:
took implementation from annotation api